pkg/util/cache.go

```diff
@@ -26,6 +19,21 @@ type Cache[T any] struct {
 	defaultExpiration time.Duration
 	mu                sync.RWMutex
 	data              map[string]CacheItem[T]
+	cleaner           *IntervalCacheCleaner[T]
+}
+
+func NewCache[T any](expiration, clean time.Duration) *Cache[T] {
```

Comment:
What if, to keep the underlying components separately testable, this self-cleaning cache was it's own struct? It would just be composed of both parts, the cache and the cleaner, and would provide the start and stop methods you mention.

Reply:
Yeah we can do! The reason I integrated it this way was because, we seem to always accompany a cache with a cache cleaner in the code, and so it made sense to combine them into a single entity. I guess a cache without a cleaner would need to come with a warning that it would be prone to memory leaks?